integrations/tensorflow/e2e/slim_vision_models/slim_vision_model_test.py

```diff
@@ -37,13 +37,26 @@
     'mobilenet_v2_035_224]\nAt least a subset can be viewed here:\n'
     'https://tfhub.dev/s?dataset=imagenet&module-type=image-classification,image-classifier'
 )
-flags.DEFINE_string('tf_hub_url', None,
-                    'Base URL for the models to test. URL at the time of '
-                    'writing:\nhttps://tfhub.dev/google/imagenet/')
+flags.DEFINE_string(
+    'tf_hub_url', None, 'Base URL for the models to test. URL at the time of '
+    'writing:\nhttps://tfhub.dev/google/imagenet/')
 
 # Classification mode; 4 - is a format of the model (SavedModel TF v2).
 MODE = 'classification/4'
-INPUT_SHAPE = (1, 224, 224, 3)
+LARGE_MODELS = ['amoebanet_a_n18_f448', "nasnet_large", "pnasnet_large"]
+
+
+def get_input_shape():
+  if FLAGS.model in LARGE_MODELS:
+    return (1, 331, 331, 3)
+  elif FLAGS.model.startswith('mobilenet_v2'):
+    # The MobileNetV2 models have variable size that seems to be only inferrible
+    # from their TFHub name.
+    size = int(FLAGS.model.split('_')[-1])
+    return (1, size, size, 3)
+  else:
+    # Default input shape.
+    return (1, 224, 224, 3)
```

Comment:
:-/ This feels really hacky. There's no way to ask the model what it's input size is?

Reply:
Yeah :/ I went through all of the attributes on the model object you get and through the attributes on its layers and there's nothing concrete.

The conditional structure is at least basically identical to that of `vision_model_test.py`.

Reply:
:vomiting_face: